# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/BackgroundParentImpl.cpp
# Commit: dbdca4661a35
# Full Hash: dbdca4661a35c55d343cb76fd10366cdbd1e4c09
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-18 09:51:31
# Regressor Bug: 1351231
# File Overlap Count: 2
# Description:
#   Bug 1351231 - PFetch protocol declaration and implementation. r=dom-worker-reviewers,jesup
#   
#   Depends on D138813
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142436
# ==============================================================================

diff -r ea27cd66fefd -r dbdca4661a35 ipc/glue/BackgroundParentImpl.cpp
--- a/ipc/glue/BackgroundParentImpl.cpp	Tue Jan 17 15:13:44 2023 +0000
+++ b/ipc/glue/BackgroundParentImpl.cpp	Tue Jan 17 15:13:45 2023 +0000
@@ -20,6 +20,7 @@
 #include "mozilla/dom/ContentParent.h"
 #include "mozilla/dom/DOMTypes.h"
 #include "mozilla/dom/EndpointForReportParent.h"
+#include "mozilla/dom/FetchParent.h"
 #include "mozilla/dom/FileCreatorParent.h"
 #include "mozilla/dom/FileSystemManagerParentFactory.h"
 #include "mozilla/dom/FileSystemRequestParent.h"
@@ -1485,6 +1486,10 @@
                                                                aClientId);
 }
 
+already_AddRefed<dom::PFetchParent> BackgroundParentImpl::AllocPFetchParent() {
+  return MakeAndAddRef<dom::FetchParent>();
+}
+
 already_AddRefed<mozilla::net::PWebSocketConnectionParent>
 BackgroundParentImpl::AllocPWebSocketConnectionParent(
     const uint32_t& aListenerId) {